
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Welcome to 10x demultiplexing pipeline using 5’ HTO for sample hashing with VDJ’s documentation! &#8212; 10x demultiplexing pipeline using 5&#39; HTO for sample hashing with VDJ v0.1.0 documentation</title>
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="welcome-to-10x-demultiplexing-pipeline-using-5-hto-for-sample-hashing-with-vdj-s-documentation">
<h1>Welcome to 10x demultiplexing pipeline using 5’ HTO for sample hashing with VDJ’s documentation!<a class="headerlink" href="#welcome-to-10x-demultiplexing-pipeline-using-5-hto-for-sample-hashing-with-vdj-s-documentation" title="Permalink to this headline">¶</a></h1>
<p>This is a repository to help you demultiplex your single cell RNA-seq experiments when you have used all of the following chemistries on a single library:</p>
<ul class="simple">
<li><p>10x Genomics scRNA-seq technology</p></li>
<li><p>Hash Tag Oligos (HTOs) for multiplexing samples in the same library</p></li>
<li><p>5’ VDJ sequencing of T-cell (TCR) or B-cell receptors (BCR)</p></li>
</ul>
<div class="section" id="dependencies">
<h2>Dependencies<a class="headerlink" href="#dependencies" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>cellranger &gt;=6.1.2 (tested on 6.1.2)</p></li>
<li><p>bash &gt;= v4.0</p></li>
</ul>
<p>This pipeline is dependent on the open-source code, <a class="reference external" href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/tutorial_in">cellranger</a>, provided by 10x genomics, particulary the <a class="reference external" href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/multi">multi</a> command functionality and the <a class="reference external" href="https://support.10xgenomics.com/docs/bamtofastq#header">bamtofastq</a> functionality.  If you have downloaded the full cellranger suite, <strong>multi</strong> and <strong>bamtofastq</strong> should already come pre-bundled and installed in the initial cellranger installation.</p>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>Once cellranger is installed, there are no other true installations besides cloning this repository:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">tbrunetti</span><span class="o">/</span><span class="mi">10</span><span class="n">x_scRNA_VDJ_5pirme_HTO_dumux</span>
</pre></div>
</div>
</div>
<div class="section" id="full-pipeline-overview">
<h2>Full Pipeline Overview<a class="headerlink" href="#full-pipeline-overview" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
</div>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="#">10x demultiplexing pipeline using 5' HTO for sample hashing with VDJ</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2022, Tonya Brunetti.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.5.4</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>